src/Features/Core/Portable/RemoveUnnecessaryImports/AbstractRemoveUnnecessaryImportsDiagnosticAnalyzer.cs

```diff
@@ -103,7 +129,17 @@ private void AnalyzeSemanticModel(SemanticModelAnalysisContext context)
 
                 EnsureClassificationIdDescriptors();
                 var fadeOut = workspaceOptions.Services.Workspace.Options.GetOption(FadingOptions.FadeOutUnusedImports, language);
-                var descriptor = fadeOut ? _unnecessaryClassificationIdDescriptor : _classificationIdDescriptor;
+
+                DiagnosticDescriptor descriptor;
+                var syntaxFacts = workspaceOptions.Services.GetLanguageServices(language).GetRequiredService<ISyntaxFactsService>();
```

Comment:
this will make this not possible to run on compiler?

Reply:
wondering why we do this extra thing to not show errors if it can only run on IDE? its not like it will fail build or anything?

Reply:
This is no different from any of our other current analyzers. When IDE0005 eventually moves to the Code Style layer, we'll likely have two separate diagnostic analyzers:

1. The analyzer in the code style layer will use `GeneratedCodeAnalysisFlags.None` and report IDE0005 on unused imports
2. The analyzer in the features layer will handle fading, the diagnostic that triggers the code fix, and the code fix itself